cmake/compiler/gcc.cmake

```diff
@@ -1,10 +1,18 @@
+if("${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "zephyr")
+  set(cc "gcc")
+  set(c++ "g++")
+elseif("${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "xcc")
+  set(cc "xcc")
+  set(c++ "xc++")
+endif()
```

Comment:
gcc.cmake should not make references to individual toolchains unless necessary.

In this particular case it is not necessary because one could have an overridable default for 'cc' and 'c++'.

set_ifndef(cc gcc)

(And then set(cc xcc) in xcc.cmake)

Reply:
Will make this change.